src/main/java/sim/model/Producer.java

```diff
@@ -23,6 +23,7 @@
 public class Producer implements Runnable {
     File in; // orders come from here
     SharedQueue out; // orders go here
+    SharedQueue priorityOut;
```

Comment:
I think for now, I'd just take this out and instantiate a second producer feeding the priority queue from the same file.

Remember this PR is about implementing the priority queue, best to keep it concise and then make any producer changes afterwards.

Reply:
if the case of reading from the same file, the process line would still need to modified right? 

Say if we add another boolean to `orders.csv`, would that not modify the constructor of Order? Unless the changes is made to the `orderId` of `orders.csv`?

Reply:
Nope, not at all, it would just work 😅

Reply:
Only thing I could see being an issue is two places reading from the same file at the same time 🤔 But I don't think it should be a problem

Reply:
so how does the producer know which line is priority and which isn't

Reply:
And the other problem is that would the priority queue not be process from top to bottom of the csv before normal start? 🤨

Reply:
@jonleesy It wouldn't, it would just duplicate the orders, but we don't really care about that just now this PR is just about adding a priority queue

Reply:
ah ok